packages/hash/api/src/model/util.ts

```diff
@@ -270,18 +269,25 @@ export const propertyTypeInitializer = (
   };
 };
 
+/**
+ * Some models may reference themselves. This marker is used to stop infinite loops during initialization by telling the initializer to use a self reference
+ */
+export const SELF_REFERENCE_MARKER = "SELF_REFERENCE";
```

Comment:
This is fine, but I think the more standard approach of doing this would've been to just define the type directly:
```suggestion
/**
 * Some models may reference themselves. This marker is used to stop infinite loops during initialization by telling the initializer to use a self reference
 */
export type SELF_REFERENCE_MARKER = "SELF_REFERENCE";
```

So that callers of the method don't have to import the constant, they can just pass the `"SELF_REFERENCE"` string.

Happy to merge this in as is though.

Reply:
I feel odd about magic strings but I agree this might actually be better, have added